comp/core/autodiscovery/common/utils/prometheus_kubelet.go

```diff
@@ -49,9 +53,11 @@ func ConfigsForPod(pc *types.PrometheusCheck, pod *kubelet.Pod) []integration.Co
 				log.Debugf("Container '%s' doesn't match the AD configuration 'kubernetes_container_names', ignoring it", containerStatus.Name)
 				continue
 			}
-			if containerUsingThePort != "" && containerStatus.Name != containerUsingThePort {
+
+			if hasPortAnnotation && containerStatus.Name != containerWithPortInAnnotation {
```

Comment:
can we return an error in case of name mismatch so it can be reported in the `agent status` and in the futur in Fleet Autotmation?